qatrfm/cli.py

```diff
@@ -44,35 +65,36 @@ def print_version(ctx, param, value):
               'comas of the Class(es) in path to be executed.')
 @click.option('--path', '-p', required=True,
               help='Path of the test file.')
-@click.option('--image', '-i', required=True, help='Path to source image.')
-@click.option('--num_domains', '-n', default=1,
-              help='Number of domains to be created. Default=1')
-@click.option('--cores', '-c', default=1, help='Num cores of the domains. '
-              'Default=1')
-@click.option('--ram', '-r', default=1024, help='Ram of the domains in MB. '
-              'Default=1024')
-@click.option('--snapshots', '-s', is_flag=True,
+@click.option('--tfvar', type=str, multiple=True, help='Variable to '
+              'insert to the .tf file. It can be used multiple times '
+              'for each single variable. At least tfvar "image" should be '
+              'provided for the default .tf file.')
+@click.option('--snapshots', is_flag=True,
               help='Create snapshots of the domains at the beginning. '
               'This is useful to allow the test revert the domains to their '
               'initial state if needed.')
 @click.option('--no-clean', 'no_clean', is_flag=True,
               help="Don't clean the environment when the tests finish. "
               "This is useful for debug and troubleshooting.")
-def cli(test, path, image, num_domains, cores, ram, snapshots, no_clean):
+def cli(test, path, tfvar, snapshots, no_clean):
     """ Create a terraform environment and run the test(s)"""
 
     logger = QaTrfmLogger.getQatrfmLogger(__name__)
     test_array = test.split(',')
 
     basedir, filename = os.path.split(path)
-    tf_file = None
-    for file in os.listdir(basedir):
-        if file.endswith(".tf"):
-            tf_file = os.path.join(basedir, file)
-    env = TerraformEnv(image=image,
-                       tf_file=tf_file,
-                       num_domains=num_domains,
-                       snapshots=snapshots)
+
+    tf_file = find_tf_file(basedir)
+
+    if (tf_file is None):
+        path_tf = os.path.dirname(os.path.realpath(__file__))
+        tf_file = ("{}/config/default.tf".format(path_tf))
+        check_image(tfvar)
+    if (not os.path.isfile(tf_file)):
+        logger.error("File {} not found.".format(tf_file))
```

Comment:
any good reason why in case of missing .tf file you throwing FileNotFoundError and here writing log message and doing exit ? I think we should be consistent with such things  ( I would vote for exceptions as Java guy :) )

Reply:
Hm, I like a formatted error message for known errors more, then a full calltrace.

Reply:
Good catch, it's actually a mistake and should be exception as in the other case. @cfconrad The exception can contain the same message error. I will update it.

Reply:
done